week4/main.cpp

```diff
@@ -0,0 +1,73 @@
+//CodeU:
+//
+//  Created by Julia Semavina on 15.06.2017
+//
+//  problem specification:
+//      You are given a 2-dimensional map of tiles. Each tile is either land or water.
+//      You have to write a function that counts the number of islands.
+//
+//  comments:
+//      DFS
+//
+
+
+#include <iostream>
+#include <vector>
+
+
+void DFS(int x, int y, int IslandSize, std::vector<std::vector<bool>> &Island,
+         std::vector<std::vector<bool>> &Visited) {
+    Visited[y][x] = true;
+    if (x + 1 < IslandSize && Island[y][x + 1] && !Visited[y][x + 1]) { // right
```

Comment:
I think it will be more clear having a loop over the 4 cardinal directions (either using two arrays for delta_x & delta_y, or looping over all 8 directions and only using the ones you need).